runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -126,6 +125,19 @@ public Table(@Nonnull final TableParameters<K, V, M> tableParameters,
         }
     }
 
+    /**
+     * Helps determine if this table has metadata fields setup
+     * @return Metadata options if setup otherwise a default instance
+     */
+    public MetadataOptions getMetadataOptions() {
+        return Optional.ofNullable(tableParameters.getMetadataSchema())
```

Comment:
Because we don't cache the MetadataOptions - it's a memory vs cpu trade off but the difference is so minor I think it is ok to just stash the MetadataOptions - I will make this change.